qiita_plugins/biom_type/biom_type_plugin/summary.py

```diff
@@ -0,0 +1,40 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+
+def generate_html_summary(qclient, job_id, parameters, out_dir):
+    """Generates the HTML summary of a BIOM artifact
+
+    Parameters
+    ----------
+    qclient : qiita_client.QiitaClient
+        The Qiita server client
+    job_id : str
+        The job id
+    parameters : dict
+        The parameter values to validate and create the artifact
+    out_dir : str
+        The path to the job's output directory
+
+    Returns
+    -------
+    dict
+        The results of the job
+
+    Raises
+    ------
+    ValueError
+        If there is any error gathering the information from the server
+    """
+    # Only one paramater
+    # artifact_id = parameters['artifact']
+    # TODO
+    # Step 1: gather file information from qiita using REST api
+    # Step 2: generate HTML summary
+    # Step 3: add the new file to the artifact using REST api
+    pass
```

Comment:
should this stub still conform to the described signature (e.g., raise `ValueError` or return `dict`)?